parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -234,18 +237,53 @@ public Encoding getEncoding(org.apache.parquet.column.Encoding encoding) {
     return Encoding.valueOf(encoding.name());
   }
 
-  public static Statistics toParquetStatistics(org.apache.parquet.column.statistics.Statistics statistics) {
+  /**
+   * @deprecated Replaced by {@link #toParquetStatistics(
+   * String createdBy, org.apache.parquet.column.statistics.Statistics statistics)}
+   */
+  @Deprecated
+  public static Statistics toParquetStatistics(
+      org.apache.parquet.column.statistics.Statistics statistics) {
+    return toParquetStatistics(null, statistics);
+  }
+
+  public static Statistics toParquetStatistics(
+      String createdBy, org.apache.parquet.column.statistics.Statistics statistics) {
     Statistics stats = new Statistics();
     if (!statistics.isEmpty()) {
       stats.setNull_count(statistics.getNumNulls());
-      if(statistics.hasNonNullValue()) {
+      if (statistics.hasNonNullValue() &&
+          createdBy != null &&
+          !shouldIgnoreStatistics(createdBy)) {
         stats.setMax(statistics.getMaxBytes());
         stats.setMin(statistics.getMinBytes());
-     }
+      }
     }
     return stats;
   }
 
+  private static boolean shouldIgnoreStatistics(String createdBy) {
+    if (!createdBy.startsWith("parquet-mr")) {
+      return false; // Assume that other implementations got the statistics correct
+    }
+
+    return parseVersion(createdBy) < STATISTICS_FIXED_VERSION;
```

Comment:
It will work for the case you are referring to as it takes first five characters of version string. However, the current logic is brittle. Will improve it.